### DISCUSSION TITLE
Traffic Light Dataset. Different bounding boxes with different APIs??

### DISCUSSION THREAD

**Discussion by lsangreg** [carla-simulator/carla#6135] [Q&A]
Hello, I am experiencing an issue with bounding boxes and traffic light.
I obtain correct bounding boxes around traffic light when I compute them as 
```
bounding_box_set = self.world.get_level_bbs(carla.CityObjectLabel.TrafficLight)
ego_position = vehicle.get_transform().location
for bb in bounding_box_set:
    dist=bb.location.distance(ego_position) 
    if dist <= 50:
         verts = [v for v in bb.get_world_vertices(carla.Transform()) 
 ```
 While if I obtain bounding boxes as 
 ```
  traffic_light_list= actor_list.filter('traffic.traffic_light*')
  ego_position = vehicle.get_transform().location
  for tl in traffic_light_list:
     bb = tl.bounding_box
     dist=bb.location.distance(ego_position) 
     if dist <= 50:
         verts = [v for v in bb.get_world_vertices(tl.get_transform())]***
```
they are wrong and different.
 I need to use the second method because I want to retrieve also the state of the traffic light, information that I cannot access to if I cycle through the bounding_box_set.
 ***P.S.: I tried using, instead of `tl.get_transform()` also `Carla.Transform()` and `ego_vehicle.get_transform()`; and instead of  `bb.get_world_vertices` I tried `bb.get_local_vertices()` but no one of this methods was working.

**Comment by cuteboyqq** [comment]
does this code able to get 2D bounding box information and draw 2D bounding box on pygame ?

**Reply by lsangreg** [reply]
Not on pygame, you can find the whole code in the documentation of Carla regarding bounding boxes
